DungeonApi/dungeon/views.py

```diff
@@ -0,0 +1,51 @@
+from django.shortcuts import render
+
+from django.http.response import JsonResponse
+from rest_framework.parsers import JSONParser
+from rest_framework import status
+
+from dungeon.models import Npc
+from dungeon.serializers import NpcSerializer
+from rest_framework.decorators import api_view
+
+@api_view(['GET', 'POST', 'DELETE'])
+def npc_list(request):
+    if request.method == 'GET':
+        npcs = Npc.objects.all()
+
+        npc_serializer = NpcSerializer(npcs, many=True)
+        return JsonResponse(npc_serializer.data, safe=False)
+
+    elif request.method == 'POST':
+        npc_data = JSONParser().parse(request)
+        npc_serializer = NpcSerializer(data=npc_data)
+        if npc_serializer.is_valid():
+            npc_serializer.save()
+            return JsonResponse(npc_serializer.data, status=status.HTTP_201_CREATED)
+        return JsonResponse(npc_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    elif request.method == 'DELETE':
+        count = Npc.objects.all().delete()
+        return JsonResponse({'message': '{} NPCS were deleted successfully!'.format(count[0])}, status=status.HTTP_204_NO_CONTENT)
```

Comment:
Would we want to add authentication to this eventually?

Reply:
This was more for practice for us to get use to how django does api calls, but if in the future we want to use this calls then yes we will need to add authentication